src/tequila/circuit/circuit.py

```diff
@@ -965,3 +965,34 @@ def from_moments(moments: typing.List):
         """
         raise TequilaException(
             'this method should never be called from Moment. Call from the QCircuit class itself instead.')
+
+
+
+def assign_ctrl(U0: QCircuit = None, U1: QCircuit = None)->int:
```

Comment:
as this can be used for other things as well: maybe rename to "find_unused_qubits", if plural in the name it might be bumped up that it gives back a whole set of them (optional).